### ISSUE TITLE
issue with firebase when upgrading to expo sdk 52

### ISSUE THREAD

**Issue by ViljamHaanpaa** [expo/expo#32429]
### Minimal reproducible example

npm install react-native-firebase/app then npm install expo@next then npx expo install --fix

### What platform(s) does this occur on?

iOS

### Where did you reproduce the issue?

in a development build

### Summary

i get this error in console 
![image](https://github.com/user-attachments/assets/5cfc767e-6492-4b92-898a-1cb015d33acf)

### Environment

```text
expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 11 10.0.22631
    Binaries:
      Node: 20.14.0 - C:\Program Files\nodejs\node.EXE
      npm: 9.7.2 - C:\Program Files\nodejs\npm.CMD
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
cant because expo-doctor tries to upgrade other depencies but it fails cos firebase
```

**Comment by timoisalive** [comment]
What about when building in EAS, is there an option to choose `--legacy-peer-deps`?

**Comment by contactsimonwilson** [comment]
> What about when building in EAS, is there an option to choose `--legacy-peer-deps`?

Yes, you can do it with the .npmrc file with legacy-peer-deps=true
